test/integration/smoke_test.go

```diff
@@ -0,0 +1,152 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	composectl "github.com/foundriesio/composeapp/cmd/composectl/cmd"
+	"os"
+	"os/exec"
+	"path"
+	"testing"
+)
+
+var (
+	composeExec = os.Getenv("COMPOSECTL_EXE")
+)
+
+func TestSmoke(t *testing.T) {
+	appName := "app"
+	appBaseUri := "registry:5000/factory/" + appName
+	appComposeDef := `
+services:
+  busybox:
+    image: ghcr.io/foundriesio/busybox:1.36
+    command: sh -c "while true; do sleep 60; done"
+`
+	appDir := path.Join(t.TempDir(), appName)
+	digestFile := path.Join(t.TempDir(), "app.sha256")
+	err := os.MkdirAll(appDir, 0o755)
+	if err != nil {
+		t.Fatal(err)
+	}
+	err = os.WriteFile(path.Join(appDir, "docker-compose.yml"), []byte(appComposeDef), 0o640)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	runCmd := func(t *testing.T, args ...string) []byte {
+		c := exec.Command(composeExec, args...)
+		c.Dir = appDir
+		output, err := c.CombinedOutput()
+		if err != nil {
+			t.Errorf("failed to run `%s` command: %s\n", args[0], output)
```

Comment:
You probably want `t.Fatal` here or your test is just going to fail everything after this. It's also going to get tedious checking for errors everywhere. The `require.Nil` package/function makes this a lot easier

Reply:
I get the feeling all your tests are going to need this function. I'm guessing the code is going to be easier if you make this a function

Reply:
Let me restructure the PR so the fixtures appear at first and then the test cases. Probably, after that the comments will become irrelevant.

Reply:
>The require.Nil package/function makes this a lot easier

Agree, the simple function should do the trick, not sure it is worth pulling another deps for it though.

Reply:
@doanac I've restructured commits in the PR. Please, review it now.